platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/signed/SignedState.java

```diff
@@ -309,6 +309,19 @@ void incrementReservationCount(@NonNull final String reason, final long reservat
         reservations.reserve();
     }
 
+    /**
+     * Try to increment the reservation count.
+     */
+    boolean tryIncrementReservationCount(@NonNull final String reason, final long reservationId) {
```

Comment:
I'd suggest adding a short unit test to verify this function's behavior. (OK if it doesn't test race conditions, just that the reservation count is mutated as expected.)

Reply:
all the tests that exist today verify this behavior through the ReservedSignedStateTests. I added a test there which tests this method as well.